File: ObjectQueue/StudentQueue.java
package ObjectQueue;
import java.io.IOException;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

// Fila - FIFO - First In, First Out

public class StudentQueue {
    public static void main(String[] args){
        Scanner keyboard = new Scanner(System.in);
        Queue<Student> StQueue = new LinkedList<Student>();

        int Registration;
        String Name;

        for(int i = 0; i < 3; i ++){
            System.out.println("Registration: ");
            Registration = keyboard.nextInt();
            keyboard.nextLine();
            System.out.println("Name: ");
            Name = keyboard.nextLine();
            System.out.println();
            StQueue.add(new Student(Registration, Name));
            
        }

        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println("------------------------------------------------------------------------------");

        System.out.println("Queue: ");
        for (Student aluno : StQueue) {
            System.out.println(aluno.toString());
        }

        System.out.println("Showing head: ");
        System.out.println(StQueue.peek());
        System.out.println();

        System.out.println("Is Queue empty?");
        System.out.println(StQueue.isEmpty());
        System.out.println();

        Student a = new Student(5555, "Angela");
        System.out.println("Checking if 'Angela' is in the Queue: " + StQueue.contains(a));
        System.out.println();
        
        System.out.println("Queue size: " + StQueue.size());
        System.out.println();

        while(!StQueue.isEmpty())
            System.out.println("Removing " + StQueue.poll()); /*poll */

        keyboard.close();
    }
}

File: Fila/ObjectQueue/Python/Student_Queue.py
import queue
import time
from student import Student

student_queue = queue.Queue()

student_queue.put(1, "Angela")
student_queue.put(2, "Pedro")
student_queue.put(3, "Joao")

print("Exibindo fila: ")
for item in student_queue.queue:
    print(item)

print("Showing head: ")
print(student_queue.queue[0])

print("Showing end of queue: ")
print(student_queue.queue[-1])

print("Checking if queue is empty: ")
print(student_queue.empty())

print("Size: ")
print(student_queue.qsize())    

print("Checking if i am in the queue: ")
a = Student(1,"Angela")
print(a in student_queue.queue)

while (not student_queue.empty()):
    print("Removing ...")
    print(str(student_queue.get_nowait()))
    time.sleep(1)